src/client/interpreter/activation/types.ts

```diff
@@ -10,7 +10,7 @@ export const IEnvironmentActivationService = Symbol('IEnvironmentActivationServi
 export interface IEnvironmentActivationService {
     getActivatedEnvironmentVariables(
         resource: Resource,
-        interpreter?: PythonEnvironment,
+        interpreter: PythonEnvironment,
```

Comment:
Its never undefined, hence (all of these optionals make the code lot more messy).
Fortunately most of them aren't optional anymore (simply because we were always passing the values).